pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -622,6 +637,8 @@ impl LayerInner {
 
                 // check if we really need to be downloaded; could have been already downloaded by a
                 // cancelled previous attempt.
+                //
+                // FIXME: what if it's a directory? that is currently needs_download == true
```

Comment:
It took me a minute to work out what this comment meant in this context: it probably would be better placed in needs_download (and return an io::Error if we see a directory somewhere we expected a file).

Reply:
Hmm makes sense actually, why did I not think about that. I'll fix it in a separate PR.